Assets/Scripts/Guns/GunController.cs

```diff
@@ -62,6 +62,15 @@ private void HandleFire()
       }
     }
 
+    private void DisableMovementGuns(IGun gunNotToDisable)
+    {
+      foreach (var gun in _guns.Values)
+      {
+        if (gun != gunNotToDisable && gun is IMovementGun)
```

Comment:
You can do "gun is IMovementGun movementGun" and use the movementGun instead of casting it the second time